src/azure-cli-core/azure/cli/core/test_utils/vcr_test_base.py

```diff
@@ -264,12 +264,12 @@ def _before_record_request(request):
         request.uri = re.sub('/sig=([^/]+)&', '/sig=0000&', request.uri)
         request.uri = _scrub_deployment_name(request.uri)
         # remove randomized portion from resource group name
-        resource_group_name = re.search('resource[Gg]roups/([^/]+_)/', request.uri)
+        resource_group_name = re.search('resource[Gg]roups/([^/^?]+_)', request.uri)
```

Comment:
You are correct that it doesn't fix the names in the body. However, I have not had that cause an issue. I do have a fix in the branch I'm working on where it doesn't currently scrub resourceGroup name that is included as part of a query string $filter.